plugin.go

```diff
@@ -0,0 +1,186 @@
+	"aspect.build/cli/pkg/bazel"
+	"aspect.build/cli/pkg/ioutils"
+	"aspect.build/cli/pkg/plugin/sdk/v1alpha4/config"
+	aspectplugin "aspect.build/cli/pkg/plugin/sdk/v1alpha4/plugin"
+)
+
+// main starts up the plugin as a child process of the CLI and connects the gRPC communication.
+func main() {
+	goplugin.Serve(config.NewConfigFor(&LintPlugin{}))
+}
+
+// LintPlugin declares the fields on an instance of the plugin.
+type LintPlugin struct {
+	// Base gives default implementations of the plugin methods, so implementing them below is optional.
+	// See the definition of aspectplugin.Base for more methods that can be implemented by the plugin.
+	aspectplugin.Base
+	// This plugin will store some state from the Build Events for use at the end of the build.
+	command_line.CommandLine
+}
+
+// CustomCommands contributes a new 'lint' command alongside the built-in ones like 'build' and 'test'.
+func (plugin *LintPlugin) CustomCommands() ([]*aspectplugin.Command, error) {
+	return []*aspectplugin.Command{
+		aspectplugin.NewCommand(
+			"lint",
+			"Run configured linters over the dependency graph.",
+			"Run linters and collect the reports they produce. TODO: more usage docs",
+			func(ctx context.Context, args []string, bazelStartupArgs []string) error {
+				streams := ioutils.DefaultStreams
+
+				// Build with the linter aspect collecting the 'report' output group.
+				// TODO: list of linter aspects should come from config
+				bazelCmd := bazelStartupArgs
+				bazelCmd = append(bazelStartupArgs, "build", "--aspects", "//:lint.bzl%eslint", "--output_groups=report")
+				bazelCmd = append(bazelCmd, args...)
+
+				if exitCode, err := bazel.WorkspaceFromWd.RunCommand(streams, nil, bazelCmd...); exitCode != 0 {
+					fmt.Fprintf(streams.Stderr, "ERROR(bazel build --aspects): %v\n", err)
+
+					err = &aspecterrors.ExitError{
+						Err:      err,
+						ExitCode: exitCode,
+					}
+					return err
+				}
+
+				// Find the lint result files.
+				lintFiles, err := plugin.findLintResultFiles(streams, bazelStartupArgs)
+				if err != nil {
+					fmt.Fprintf(streams.Stderr, "FIND lint results: %v\n", err)
+
+					err = &aspecterrors.ExitError{
+						Err:      err,
+						ExitCode: 1,
+					}
+					return err
+				}
+
+				if len(lintFiles) == 0 {
+					return nil
+				}
+
+				// Output the lint results.
+				for _, f := range lintFiles {
+					lintResultBuf, err := os.ReadFile(f)
+					if err != nil {
+						fmt.Fprintf(streams.Stderr, "READ(%q) result failure: %v\n", f, err)
+
+						err = &aspecterrors.ExitError{
+							Err:      err,
+							ExitCode: 1,
+						}
+						return err
+					}
+
+					lineResult := strings.TrimSpace(string(lintResultBuf))
+					if len(lineResult) > 0 {
+						fmt.Fprintln(streams.Stderr, lineResult)
+					}
+				}
+
+				return nil
+			},
+		),
+	}, nil
+}
+
+func (plugin *LintPlugin) findLintResultFiles(streams ioutils.Streams, bazelStartupArgs []string) ([]string, error) {
+	// TODO: use bazel query
```

Comment:
actually the TODO should be to watch the build event stream as it goes by, I expect linting actions along with their output paths (including for cache hits) will be reported.